src/main/java/ch/njol/skript/lang/Variable.java

```diff
@@ -122,7 +124,15 @@ public static boolean isValidVariableName(String name, final boolean allowListVa
 				Skript.error("A variable's name must neither start nor end with the separator '" + SEPARATOR + "' (error in variable {" + name + "})");
 			return false;
 		} else if (name.contains("*") && (!allowListVariable || name.indexOf("*") != name.length() - 1 || !name.endsWith(SEPARATOR + "*"))) {
-			if (printErrors) {
+			int count = StringUtils.count(name, '*');
+			Matcher m = Pattern.compile("(%(.*?)%)+").matcher(name); // Matches all data within %%
```

Comment:
You shouldn't be compiling a pattern each time the method is called, it's better to have this in a constant and apply the matcher later.